src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_vm_utils.py

```diff
@@ -46,6 +46,19 @@ def _resolve_api_version(provider_namespace, resource_type, parent_path):
             .format(resource_type))
 
 
+def log_pprint_template(template):
+    import azure.cli.core.azlogging as azlogging
+    from pprint import pprint
+    import sys
+    from six import StringIO
+
+    logger = azlogging.get_az_logger(__name__)
+    io = StringIO()
+    pprint(template, io)
+    logger.warning(io.getvalue())
```

Comment:
Yep, json.dumps looks better. Also, I changed the logging level to info so you have to use --verbose to see this. I thought that was more appropriate than displaying it by default or requiring --debug.